src/liballoc/collections/btree/node.rs

```diff
@@ -73,7 +73,7 @@ struct LeafNode<K, V> {
     /// This node's index into the parent node's `edges` array.
     /// `*node.parent.edges[node.parent_idx]` should be the same thing as `node`.
     /// This is only guaranteed to be initialized when `parent` is nonnull.
-    parent_idx: u16,
+    parent_idx: MaybeUninit<u16>,
 
     /// The number of keys and values this node stores.
     ///
```

Comment:
Can you confirm that despite introducing `MaybeUninit` we are still the same size? (i.e., same 32-bit word layout as indicated by the comment on len)?

Reply:
`MaybeUninit` won't introduce any additional alignment requirements that aren't present in the type it wraps, and will only increase the size of the resulting type as a result of inhibiting niche-filling optimizations (and u16 has not niches, so this shouldn't be an issue).

Reply:
What would be a good way to test this? We have a way to "hack" a `static_assert!`, but the size alone won't do it here because there's a ptr right before this, so on 64bit system it'll be 2 ptr-words in size at least. And we cannot yet do an `offset!` macro in CTFE. I could add a `debug_assert!` to test the offset at run-time?

However, I agree with what @cramertj said and this is `repr(C)`, so I am not expecting any surprises.  Still, would be nice to be sure.